pkg/filesystem/path/builder_test.go

```diff
@@ -31,13 +32,77 @@ func TestBuilder(t *testing.T) {
 			"/hello/../world/foo",
 		} {
 			t.Run(p, func(t *testing.T) {
+				// Unix Parser
 				builder1, scopewalker1 := path.EmptyBuilder.Join(path.VoidScopeWalker)
 				require.NoError(t, path.Resolve(path.MustNewUNIXParser(p), scopewalker1))
 				require.Equal(t, p, builder1.GetUNIXString())
 
 				builder2, scopewalker2 := path.EmptyBuilder.Join(path.VoidScopeWalker)
 				require.NoError(t, path.Resolve(builder1, scopewalker2))
 				require.Equal(t, p, builder2.GetUNIXString())
+
+				// Windows Parser, compare Windows and UNIX string identity.
+				windowsStyle := strings.Replace(p, "/", "\\", -1)
+				builder3, scopewalker3 := path.EmptyBuilder.Join(path.VoidScopeWalker)
+				require.NoError(t, path.Resolve(path.MustNewWindowsParser(p), scopewalker3))
+				require.Equal(t, windowsStyle, builder3.GetWindowsString())
+				require.Equal(t, p, builder3.GetUNIXString())
+
+				builder4, scopewalker4 := path.EmptyBuilder.Join(path.VoidScopeWalker)
+				require.NoError(t, path.Resolve(builder3, scopewalker4))
+				require.Equal(t, windowsStyle, builder4.GetWindowsString())
+				require.Equal(t, p, builder4.GetUNIXString())
+			})
+		}
+
+		for _, p := range []string{
+			"/C/",
+			"/C/hello/",
+			"/C/hello/..",
+			"/C/hello/../world",
+			"/C/hello/../world/",
+			"/C/hello/../world/foo",
+			"\\C\\",
+			"\\C\\hello\\",
+			"\\C\\hello\\..",
+			"\\C\\hello\\..\\world",
+			"\\C\\hello\\..\\world\\",
+			"\\C\\hello\\..\\world\\foo",
```

Comment:
I don't think we should support these constructs. On Windows, `\C\Hello` would refer to `X:\C\Hello`, assuming `X:` is the current drive.